File: src/parser/nim/actions/status.nim
import asyncdispatch, json, strutils

import ../utils/[chalk, config]

proc nlcli_status*(s: string = "{}") {.async.} =
    let jdata = parseJSON(s)
    let enable = jdata{"enable"}.getBool()
    let disable = jdata{"disable"}.getBool()

    initconfig()

    # If no flag is supplied only print the status.
    if not enable and not disable:
        let status = getsetting("status")
        let message = if status == "1": "on".chalk("green") else: "off".chalk("red")
        echo message
    else:
        if enable and disable:
            let varg1 = "--enable".chalk("bold")
            let varg2 = "--disable".chalk("bold")
            let tstring = "? and ? given when only one can be provided."
            quit(tstring % [varg1, varg2])

        if enable:
            setsetting("status", "1")
            echo "on".chalk("green")
        elif disable:
            # let contents = `Disabled: ${new Date()};${Date.now()}`;
            setsetting("status", "0")
            echo "off".chalk("red")

File: src/parser/js2/actions/status.js
"use strict";

const chalk = require("chalk");
const { exit, fmt } = require("../utils/toolbox.js");
const { initconfig, getsetting, setsetting } = require("../utils/config.js");

module.exports = async (args) => {
	let { enable, disable } = args;

	await initconfig();

	// If no flag is supplied only print the status.
	if (!enable && !disable) {
		let status = await getsetting("status");
		let message = status === "1" ? chalk.green("on") : chalk.red("off");
		console.log(message);
	} else {
		if (enable && disable) {
			let varg1 = chalk.bold("--enable");
			let varg2 = chalk.bold("--disable");
			let tstring = "? and ? given when only one can be provided.";
			exit([fmt(tstring, varg1, varg2)]);
		}

		if (enable) {
			await setsetting("status", 1);
			console.log(chalk.green("on"));
		} else if (disable) {
			// let contents = `Disabled: ${new Date()};${Date.now()}`;
			await setsetting("status", 0);
			console.log(chalk.red("off"));
		}
	}
};
